examples/vision/transformer_in_transformer.py

```diff
@@ -0,0 +1,328 @@
+"""
+Title: Image classification with TNT(Transformer in Transformer)
+Author: [ZhiYong Chang](https://github.com/czy00000)
+Date created: 2021/10/25
+Last modified: 2021/11/06
+Description: Implementing the Transformer in Transformer (TNT) model for image classification.
+"""
+
+"""
+## Introduction
+This example implements the [TNT](https://arxiv.org/abs/2103.00112)
+model for image classification, and demonstrates it on the CIFAR-100 dataset.
+TNT is a novel model for modeling both patch-level and pixel-level
+representation. In each TNT block, an ***outer*** transformer block is utilized to
+process
+patch embeddings, and an ***inner***
+transformer block extracts local features from pixel embeddings. The pixel-level
+feature is projected to the space of patch embedding by a linear transformation layer
+and then added into the patch.
+This example requires TensorFlow 2.5 or higher, as well as
+[TensorFlow Addons](https://www.tensorflow.org/addons/overview) package,
+which can be installed using the following command:
+```python
+pip install -U tensorflow-addons
+```
+"""
+
+"""
+## Setup
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import math
+import tensorflow as tf
+import tensorflow_addons as tfa
+from tensorflow import keras
+from tensorflow.keras import layers
+from itertools import repeat
+
+
+"""
+## Prepare the data
+"""
+
+num_classes = 100
+input_shape = (32, 32, 3)
+
+(x_train, y_train), (x_test, y_test) = keras.datasets.cifar100.load_data()
+y_train = keras.utils.to_categorical(y_train, num_classes)
+y_test = keras.utils.to_categorical(y_test, num_classes)
+print(f"x_train shape: {x_train.shape} - y_train shape: {y_train.shape}")
+print(f"x_test shape: {x_test.shape} - y_test shape: {y_test.shape}")
+
+"""
+## Configure the hyperparameters
+"""
+
+weight_decay = 0.0001
+learning_rate = 0.001
+label_smoothing = 0.1
+validation_split = 0.2
+batch_size = 128
+image_size = 96  # resize images to this size
+num_epochs = 50
+
+"""
+## Use data augmentation
+"""
+
+data_augmentation = keras.Sequential(
```

Comment:
It is a little weird you are using a sequential model here when elsewhere you are using

```
some_function(input):
  return output
```

Applied to function model inputs. Why not just do the same here, and define a function which applies these layers to some functional input?